Spec/RealtimeClientConnection.swift

```diff
@@ -23,6 +23,35 @@ class RealtimeClientConnection: QuickSpec {
 
     override func spec() {
         describe("Connection") {
+            
+            // CD2c
+            context("ConnectionDetails") {
+                it("maxMessageSize overrides the default maxMessageSize") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    let client = ARTRealtime(options: options)
+                    client.setTransport(TestProxyTransport.self)
+                    let defaultMaxMessageSize = ARTDefault.maxMessageSize()
+                    expect(defaultMaxMessageSize).to(equal(65536))
+                    defer {
+                        ARTDefault.setMaxMessageSize(defaultMaxMessageSize)
+                        client.dispose()
+                        client.close()
+                    }
+                    ARTDefault.setMaxMessageSize(1)
+                    
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.connected) { _ in
+                            let transport = client.transport as! TestProxyTransport
+                            let firstConnectionDetails = transport.protocolMessagesReceived.filter{ $0.action == .connected }[0].connectionDetails
+                            expect(firstConnectionDetails!.maxMessageSize).to(equal(16384)) // Sandbox apps have a 16384 limit
```

Comment:
This is currently failing, right? Because I don't see any change where the connection is sending the current _MaxMessageSize_.

Reply:
No that's passing. And once [this](https://github.com/ably/ably-common/pull/29/files#diff-35f9a23050dc9d9017554ed78cb4e77eR6) is merged, we can be sure of that value.

Reply:
We are already loading that value https://github.com/ably/ably-ios/blob/develop/Source/ARTConnectionDetails.m#L24 It's just not exposed yet in realtime.